[torchelastic] add timing events to different stages of rendezvous

suo opened the pull request:
Summary: as title

Test Plan: unit tests. Launched a test job and observed scuba results: {F1506543300}

Differential Revision: D57018103




cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k

d4l3k reviewed (approved):
LGTM! TIL time.perf_counter()

suo committed:
[torchelastic] add timing events to different stages of rendezvous (#125636)

Summary:

as title

Test Plan: unit tests. Launched a test job and observed scuba results: {F1506543300}

Reviewed By: d4l3k

Differential Revision: D57018103
